### PR TITLE
PWX-36622: Optimize stork image by removing unused contents from google-cloud-sdk.

### PR DESCRIPTION
**Pull request by diptiranjanpx**
Signed-Off-By: Diptiranjan

**What type of PR is this?**
improvement

**What this PR does / why we need it**:
Reducing stork image size by removing unused  google-cloud-sdk contents.

**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:

```release-note
Issue: Stork image size was too big (appx. 3.8G).
User Impact: Image download was taking longer time.
Resolution: Optimized the stork image size to 1.64G.

```

**Does this change need to be cherry-picked to a release branch?**:
yes, 24.2.0

**Test**:
https://jenkins.pwx.dev.purestorage.com/job/Stork/view/Stork%20Cloud%20Jobs/job/stork-gke-migration/27/

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -35,19 +35,24 @@ RUN asdf global kubelogin latest
...
     tar xf $GCLOUD_SDK -C $GCLOUD_INSTALL_DIR && rm -rf $GCLOUD_SDK && \
     rm -rf $GCLOUD_INSTALL_DIR/google-cloud-sdk/platform/gsutil \
-    $GCLOUD_INSTALL_DIR/google-cloud-sdk/RELEASE_NOTES
-ENV PATH "${PATH}:$GCLOUD_INSTALL_DIR/google-cloud-sdk/bin"
-#Install gke-gcloud-auth-plugin
-RUN gcloud components install gke-gcloud-auth-plugin
+    $GCLOUD_INSTALL_DIR/google-cloud-sdk/RELEASE_NOTES && \
+    gcloud components install gke-gcloud-auth-plugin && \
+    mv /usr/lib/google-cloud-sdk/bin google-cloud-sdk/ && \
+    rm -rf /usr/lib/google-cloud-sdk && \
```
Comment (pp511) [inline Dockerfile:47]: All the gains came from just removing the non "bin" component of the google-cloud-sdk right?

**Review by pp511 — APPROVED** [review]
lgtm. Just some qs for my understanding.

File: Dockerfile [hunk 2]
```diff
@@ -35,19 +35,24 @@ RUN asdf global kubelogin latest
...
+# and remove the tar bundle and all directory contents except bin directory  
 RUN curl -q -o $GCLOUD_SDK https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/$GCLOUD_SDK && \
     tar xf $GCLOUD_SDK -C $GCLOUD_INSTALL_DIR && rm -rf $GCLOUD_SDK && \
     rm -rf $GCLOUD_INSTALL_DIR/google-cloud-sdk/platform/gsutil \
-    $GCLOUD_INSTALL_DIR/google-cloud-sdk/RELEASE_NOTES
-ENV PATH "${PATH}:$GCLOUD_INSTALL_DIR/google-cloud-sdk/bin"
-#Install gke-gcloud-auth-plugin
-RUN gcloud components install gke-gcloud-auth-plugin
+    $GCLOUD_INSTALL_DIR/google-cloud-sdk/RELEASE_NOTES && \
+    gcloud components install gke-gcloud-auth-plugin && \
```
[the remark this answers is not in the document]
Comment (diptiranjanpx) [reply Dockerfile:45]: If everything is in same layer it gives the benefit. So if we the size increases in one layer and then in next layer we delete, then the previous layer will anyway add up to container image size which needs to be uploaded/downloaded.

File: Dockerfile (hunk 1 above)
Comment (diptiranjanpx) [reply Dockerfile:47]: yes

**Pull request merged**